### PR TITLE
Make min accuracy more intuitive

### PR DESCRIPTION
**Pull request by JesseCSlater**
### Description

I have changed min accuracy mode to have three choices: off, standard, and strict. 
In strict mode, if your accuracy ever falls below the target accuracy, you instantly fail. (The calculation is now done every time a key is pressed instead of based on the timer.
In standard mode, there is a bit of grace given. Grace is given in two ways:
1. Making a single mistake on a test will never fail you. You must make at least two mistakes to fail. 
2. If you drop below your target accuracy, but getting the next word (next 5 characters) correct would bring you back above, you do not fail.

### CODE REVIEW

File: frontend/.firebase-example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+	"projects": {
+		"default": "monkeytype-f6501"
```
Comment (Bruception) [inline frontend/.firebase-example:3]: Remove this file.

File: frontend/package.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
     "crypto-browserify": "3.12.0",
     "damerau-levenshtein": "1.0.8",
     "date-fns": "2.28.0",
-    "firebase": "9.6.0",
+    "firebase": "^9.6.0",
```
Comment (Bruception) [inline frontend/package.json:65]: What’s up with this change. Please revert.

File: frontend/src/ts/constants/firebase-config.t [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// For Firebase JS SDK v7.20.0 and later, measurementId is optional
+const firebaseConfig = {
```
Comment (Bruception) [inline frontend/src/ts/constants/firebase-config.t:2]: What’s up with this file change?

**Comment by JesseCSlater** [comment]
I think I have fixed all the accidental changes.

File: frontend/.firebase-example (hunk 1 above)
Comment (JesseCSlater) [reply frontend/.firebase-example]: Done

File: frontend/package.json (hunk 2 above)
Comment (JesseCSlater) [reply frontend/package.json]: Don't know how I even did that

File: frontend/src/ts/constants/firebase-config.t (hunk 3 above)
Comment (JesseCSlater) [reply frontend/src/ts/constants/firebase-config.t]: Deleted

File: frontend/package-lock.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
         "crypto-browserify": "3.12.0",
         "damerau-levenshtein": "1.0.8",
         "date-fns": "2.28.0",
-        "firebase": "9.6.0",
```
Comment (Bruception) [inline frontend/package-lock.json:21]: You need to change this file too.

Comment (JesseCSlater) [reply frontend/package-lock.json]: I think it should be good now

File: frontend/src/ts/controllers/input-controller.ts [hunk 5]
```diff
@@ -507,6 +510,13 @@ function handleChar(char: string, charIndex: number): void {
     return;
   }
 
+  if (!thisCharCorrect && Config.minAcc != "off" && TestLogic.failByMinAcc()) {
```
Comment (Bruception) [inline frontend/src/ts/controllers/input-controller.ts:513]: Use strict equality `!==`

File: frontend/src/ts/controllers/input-controller.ts [hunk 6]
```diff
@@ -181,6 +181,9 @@ function handleSpace(): void {
         TestLogic.fail("difficulty");
         return;
       }
+      if (Config.minAcc != "off" && TestLogic.failByMinAcc()) {
```
Comment (Bruception) [inline frontend/src/ts/controllers/input-controller.ts:184]: Use strict equality `!==`

File: frontend/src/ts/test/test-logic.ts [hunk 7]
```diff
@@ -56,6 +56,19 @@ import { Auth } from "../firebase";
 
 let failReason = "";
 
+export function failByMinAcc() : boolean {
+  if (Config.minAcc == "strict") {
```
Comment (Bruception) [inline frontend/src/ts/test/test-logic.ts:60]: Use strict equality `!==`

File: frontend/src/ts/test/test-logic.ts [hunk 8]
```diff
@@ -56,6 +56,19 @@ import { Auth } from "../firebase";
...
+    return TestStats.calculateAccuracy() < Config.minAccCustom;
+  }
+  else if (Config.minAcc == "standard") {
```
Comment (Bruception) [inline frontend/src/ts/test/test-logic.ts:63]: Use strict equality `!==`

**Comment by Bruception** [comment]
I think there might be some stylistic issues (spacing, indentation, etc)

**Pull request closed without merging**